src/autopas/containers/verletClusterLists/VerletClusterCellsParticleIterator.h

```diff
@@ -0,0 +1,266 @@
+/**
+ * @file VerletClusterCellsParticleIterator.h
+ * @author jspahl
+ * @date 25.3.19
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <vector>
+#include "autopas/cells/FullParticleCell.h"
+#include "autopas/containers/ParticleContainer.h"
+#include "autopas/iterators/ParticleIterator.h"
+#include "autopas/iterators/RegionParticleIterator.h"
+
+namespace autopas {
+namespace internal {
+/**
+ * ParticleIterator class to access particles inside a VerletClusterCells container efficiently.
+ * The particles can be accessed using "iterator->" or "*iterator". The next
+ * particle using the ++operator, e.g. "++iterator".
+ * @tparam Particle Type of the particle that is accessed.
+ * @tparam ParticleCell Type of the container
+ */
+template <class Particle, class ParticleCell, bool modifiable>
+class VerletClusterCellsParticleIterator : public ParticleIteratorInterfaceImpl<Particle, modifiable> {
+  using CellVecType = std::conditional_t<modifiable, std::vector<ParticleCell>, const std::vector<ParticleCell>>;
+  using CellIteratorType = std::conditional_t<modifiable, typename std::vector<Particle>::iterator,
+                                              typename std::vector<Particle>::const_iterator>;
+  using DummyStartsType = std::conditional_t<modifiable, std::vector<size_t> *, const std::vector<size_t>>;
+  using ParticleType = std::conditional_t<modifiable, Particle, const Particle>;
+
+ public:
+  /**
+   * Constructor
+   * @param cont cells to iterate over
+   * @param dummyStarts indices of the first dummy particle for each cell
+   * @param offsetToDummy offset to add to create dummy particle
+   * @param behavior iterator behavior
+   */
+  explicit VerletClusterCellsParticleIterator(CellVecType *cont, DummyStartsType dummyStarts, double offsetToDummy,
+                                              IteratorBehavior behavior = haloAndOwned)
+      : _vectorOfCells(cont), _dummyStarts(dummyStarts), cellId(0), _behavior(behavior), _offsetToDummy(offsetToDummy) {
+    cellIter = (*_vectorOfCells)[0]._particles.begin();
+    cellEnd = cellIter + getDummyStartbyIndex(0);
+    --cellIter;
+
+    VerletClusterCellsParticleIterator<Particle, ParticleCell, modifiable>::operator++();
+  }
+
+  /**
+   * @copydoc ParticleIteratorInterface::operator++()
+   */
+  inline VerletClusterCellsParticleIterator<Particle, ParticleCell, modifiable> &operator++() override {
+    do {
+      ++cellIter;
+
+      while (cellIter == cellEnd) {
+        ++cellId;
+
+        if (not isValid()) {
```

Comment:
```suggestion
        if (not cellId < _vectorOfCells->size()) {
```
use this directly to make it a bit clearer, what happens here